system-test/translate.ts

```diff
@@ -135,5 +136,48 @@ describe('translate', () => {
         supportTarget: true,
       });
     });
+
+    it('should populate x-goog-user-project header, if quota_project_id provided in credentials', async () => {
+      const {GoogleAuth} = require('google-auth-library');
+      const auth = new GoogleAuth({
+        credentials: Object.assign(
+          require(process.env.GOOGLE_APPLICATION_CREDENTIALS || ''),
+          {
+            quota_project_id: process.env.GCLOUD_PROJECT,
+          }
+        ),
+      });
+      const {TranslationServiceClient} = http2spy.require(
+        require.resolve('../src')
+      );
+      const translate = new TranslationServiceClient({
+        auth,
+      });
+
+      // We run the same test as "list of supported languages", but with an
+      // alternate "quota_project_id" set; Given that GCLOUD_PROJECT
+      // references a valid project, we expect success:
+      const projectId = await translate.getProjectId();
+      const [result] = await translate.getSupportedLanguages({
+        parent: `projects/${projectId}`,
+      });
+      const englishResult = result.languages!.filter(
+        (l: {[key: string]: string}) => {
+          return l.languageCode === 'en';
+        }
+      )[0];
+      assert.deepStrictEqual(englishResult, {
+        languageCode: 'en',
+        displayName: '',
+        supportSource: true,
+        supportTarget: true,
+      });
+
+      // Ensure we actually populated the header:
+      assert.strictEqual(
+        'long-door-651',
```

Comment:
this will fail locally, i think? and only pass on kokoro.

Reply:
shoot, this should be an environment variable good catch.